### ISSUE TITLE
azure site list fails

### ISSUE THREAD

**Issue by SandyChapman** [Azure/azure-xplat-cli#1903]
Platform: OSX 10.10.4

```
$ azure --version
0.9.5 (node: 0.10.23)
```

Command execution:

```
$ azure site list
info:    Executing command site list
+ Getting locations                                                            
\ Getting locationserror:   Cannot read property 'length' of undefined         
info:    Error information has been recorded to /Users/schapman/.azure/azure.err
error:   site list command failed
```

Log file:

```
$ cat /Users/schapman/.azure/azure.err 
Wed Jul 08 2015 15:38:10 GMT-0300 (ADT):
[TypeError: Cannot read property 'length' of undefined]
TypeError: Cannot read property 'length' of undefined
    at /usr/local/azure/lib/commands/asm/site._js:72:21
    at Interactor.__.extend.formatOutput (/usr/local/azure/lib/util/interaction.js:248:7)
    at AzureCli.<anonymous> (/usr/local/azure/lib/commands/asm/site._js:71:27)
    at AzureCli.<anonymous> (/usr/local/azure/lib/cli.js:216:12)
    at AzureCli.<anonymous> (/usr/local/azure/node_modules/commander/lib/commander.js:235:8)
    at AzureCli.EventEmitter.emit (events.js:117:20)
    at AzureCli.Command.parseArgs (/usr/local/azure/node_modules/commander/lib/commander.js:405:12)
    at AzureCli.Command.parse (/usr/local/azure/node_modules/commander/lib/commander.js:351:15)
    at AzureCli.<anonymous> (/usr/local/azure/lib/cli.js:510:20)
    at AzureCli.EventEmitter.emit (events.js:95:17)
```

**Comment by NaveenGoli** [comment]
We are aware of this issue and investigating,

In the meantime, switch to ARM based commands,

Azure config mode arm

Webapp list with resource group..


Sent from my cool Windows Phone
________________________________

**Comment by amarzavery** [comment]
@SandyChapman - How did you install the CLI (from npm or the mac installer)? 

I suspect, it is the mac installer.

Can you uninstall the mac installer ```sudo azure-uninstall```

and install from npm
```npm install -g azure-cli```

Hopefully this should unblock you.

**Comment by SandyChapman** [comment]
@amarzavery That fixed it. Thanks! Feel free to resolve this (or keep it open if you plan on fixing the OSX installer before resolving this ticket).

**Comment by yugangw-msft** [comment]
I have understood the issue and have a fix for it #1908 
This is a fallout of performance work for general command loading.
